tests/generative/variable_strategies.py

```diff
@@ -1,6 +1,9 @@
 import datetime
+from functools import wraps
+from os import environ
 
 import hypothesis.strategies as st
+from hypothesis.control import current_build_context
```

Comment:
Oh nice! I didn't realise there was a non-hacky way of accessing this state. How did you find this? I couldn't seem to find anything in the docs.

Reply:
I googled something like "hypothesis test context" to try and get some way of tracking state via the frame, and found it as a happy coincidence